microagents/knowledge/forgejo.md

```diff
@@ -0,0 +1,123 @@
+---
+name: forgejo
+type: knowledge
+version: 1.0.0
+agent: CodeActAgent
+triggers:
+- forgejo
+- codeberg
+---
+
+# Forgejo Knowledge
+
+Forgejo is a self-hosted Git service that is a fork of Gitea. It provides a lightweight, open-source alternative to GitHub and GitLab. Codeberg.org is a popular instance of Forgejo. OpenHands provides robust integration with Forgejo instances, with full support for repository operations, issue tracking, and code review.
+
+## API
+
+The Forgejo API is compatible with the Gitea API. It follows a RESTful design and uses JSON for data exchange. The base URL for the API is typically:
+
+```
+https://[forgejo-instance]/api/v1
+```
+
+For Codeberg, the base URL is:
+
+```
+https://codeberg.org/api/v1
+```
+
+### ActivityPub API
+
+Forgejo also includes ActivityPub API endpoints, which allow it to federate with other ActivityPub-compatible services in the Fediverse. This is a unique feature that distinguishes Forgejo from many other Git hosting platforms.
+
+The ActivityPub API endpoints include:
+
+- `/api/v1/activitypub/user/{username}` - User's ActivityPub profile
+- `/api/v1/activitypub/user/{username}/inbox` - User's ActivityPub inbox
+- `/api/v1/activitypub/repository/{owner}/{repo}` - Repository's ActivityPub profile
+- `/api/v1/activitypub/repository/{owner}/{repo}/inbox` - Repository's ActivityPub inbox
+
+This allows Forgejo instances to federate with other Forgejo instances and interact with other ActivityPub-compatible services like Mastodon.
+
+### Authentication
+
+Forgejo supports several authentication methods:
+
+1. HTTP basic authentication
+2. `token=...` parameter in URL query string
+3. `access_token=...` parameter in URL query string
+4. `Authorization: token ...` header in HTTP headers
+
+### Common Endpoints
+
+- `/user` - Get authenticated user information
+- `/user/repos` - List repositories for the authenticated user
+- `/repos/search` - Search for repositories
+- `/repos/{owner}/{repo}` - Get repository information
+- `/repos/{owner}/{repo}/issues` - List issues for a repository
+- `/repos/{owner}/{repo}/pulls` - List pull requests for a repository
+
+### Pagination
+
+The API supports pagination with the `page` and `limit` parameters. The `Link` header is returned with next, previous, and last page links if there are more than one page.
+
+## Differences from GitHub/GitLab
+
+- Forgejo uses `limit` instead of `per_page` for pagination
+- Forgejo uses `stars_count` instead of `stargazers_count` for repository stars
```

Comment:
```suggestion
- Forgejo uses `limit` instead of `per_page` for pagination
```

Some of these instructions may be unnecessary? Could be helpful to cull the information down to things that the agent would require pointers on to work well in most agent use cases

Reply:
star count is sometimes relevant for search&local sort purposes